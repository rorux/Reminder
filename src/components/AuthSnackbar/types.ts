export interface IAuthSnackbarProps {
  message: string;
  clearError: () => void;
}
