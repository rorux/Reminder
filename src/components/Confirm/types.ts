import React from 'react';

export type TConfirmProps = {
  open: boolean;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
  handler: (event: React.MouseEvent<HTMLElement>, id: string) => void;
  id: string;
};
