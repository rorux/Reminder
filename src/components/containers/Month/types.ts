import { TRecord } from '@store/records/types';

export type TMonthProps = {
  year: number;
  month: number;
  recordList: Array<TRecord>;
};
