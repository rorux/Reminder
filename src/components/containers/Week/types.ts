import { TRecord } from '@store/records/types';

export type TWeekProps = {
  weekArray: Array<null | number>;
  recordList: Array<TRecord>;
};
