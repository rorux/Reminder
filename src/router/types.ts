import { RouteProps } from 'react-router-dom';

export interface IRouteProps extends RouteProps {
  authed: boolean | null;
}

export enum ROUTES {
  MAIN = '/',
  DASHBOARD = '/dashboard',
  CALENDAR = '/calendar',
  LOGIN = '/login',
  REGISTER = '/register',
  NOT_FOUND = '/not-found',
}
