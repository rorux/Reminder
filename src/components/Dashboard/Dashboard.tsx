import React from 'react';

const Dashboard = () => {
  return <div data-test="dashboard">Дашборд</div>;
};

export default Dashboard;
