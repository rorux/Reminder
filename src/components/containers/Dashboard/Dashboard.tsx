import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Typography from '@mui/material/Typography';
import Fab from '@mui/material/Fab';
import AddIcon from '@mui/icons-material/Add';
import Tooltip from '@mui/material/Tooltip';
import Grid from '@mui/material/Grid';
import { ROUTES } from '@router/types';
import RecordsAccordion from '@components/RecordsAccordion';
import { initRecords } from '@store/records/actions';
import { recordsSelector } from '@store/records/selectors';
import RecordListRenderer from '@utils/records/RecordListRenderer';

const Dashboard = () => {
  const { recordList } = useSelector(recordsSelector);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(initRecords());
  }, [dispatch]);

  const recordListRender = RecordListRenderer(recordList);

  return (
    <div data-test="dashboard">
      <Grid container>
        <Grid item sm>
          <Typography variant="overline" display="block" gutterBottom>
            Список дел
          </Typography>
        </Grid>
        <Grid item>
          <Link to={ROUTES.CREATE_RECORD}>
            <Tooltip title="Создать новое напоминание" placement="left">
              <Fab
                size="small"
                color="inherit"
                aria-label="add"
                sx={{ mb: 1, boxShadow: 0 }}
                variant="extended"
              >
                <AddIcon />
              </Fab>
            </Tooltip>
          </Link>
        </Grid>
      </Grid>

      {recordListRender.length ? (
        <RecordsAccordion recordListRender={recordListRender} />
      ) : (
        <Typography variant="body2" gutterBottom>
          У Вас нет сохраненных записей. Создайте напоминание, кликнув на &quot;плюс&ldquo; в правом
          верхнем углу.
        </Typography>
      )}
    </div>
  );
};

export default Dashboard;
