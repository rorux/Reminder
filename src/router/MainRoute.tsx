import React from 'react';
import { Redirect, Route } from 'react-router-dom';
import { ROUTES } from './constants';
import { IRouteProps } from './types';

const MainRoute = (props: IRouteProps) => {
  const { authed, ...rest } = props;

  return (
    <Route
      {...rest}
      render={(routeProps) =>
        authed ? (
          <Redirect
            to={{
              pathname: ROUTES.CALENDAR,
              state: { from: routeProps.location },
            }}
          />
        ) : (
          <Redirect
            to={{
              pathname: ROUTES.LOGIN,
              state: { from: routeProps.location },
            }}
          />
        )
      }
    />
  );
};

export default MainRoute;
