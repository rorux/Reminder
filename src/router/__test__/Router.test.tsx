import React from 'react';
import 'jsdom-global/register';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Route, MemoryRouter, BrowserRouter } from 'react-router-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import MyRouter from '../Router';
import PrivateRoute from '../PrivateRoute';
import PublicRoute from '../PublicRoute';
import MainRoute from '../MainRoute';
import { ROUTES } from '../types';
import Dashboard from '@components/containers/Dashboard';
import { rootStateForTesting } from '@store/types';
import PageLayout from '@screens/PageLayout';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(() => rootStateForTesting),
  useDispatch: jest.fn(),
}));

const setUp = (route: string, authed: boolean) =>
  mount(
    <MemoryRouter initialEntries={[route]}>
      <MyRouter authed={authed} />
    </MemoryRouter>
  );

describe('MainRoute', () => {
  it('Should render Calendar with auth', () => {
    const component = shallow(<MainRoute authed={true} />);
    const routeProps = { location: 'string' };
    expect(component.find(Route).props().render(routeProps).props.to.pathname).toBe(
      ROUTES.CALENDAR
    );
  });

  it('Should render Login without auth', () => {
    const component = shallow(<MainRoute authed={false} />);
    const routeProps = { location: 'string' };
    expect(component.find(Route).props().render(routeProps).props.to.pathname).toBe(ROUTES.LOGIN);
  });
});

describe('PrivateRoute', () => {
  it('Should render transmitted component with auth', () => {
    const component = shallow(<PrivateRoute authed={true} component={Dashboard} />);
    const routeProps = { location: 'string' };
    expect(component.find(Route).props().render(routeProps).type).toBe(Dashboard);
  });
  it('Should render with auth and without transmitted component', () => {
    const component = shallow(<PrivateRoute authed={true} />);
    component.find(Route).props().render();
  });
  it('Should render Login without auth', () => {
    const component = shallow(<PrivateRoute authed={false} />);
    const routeProps = { location: 'string' };
    expect(component.find(Route).props().render(routeProps).props.to.pathname).toBe(ROUTES.LOGIN);
  });
});

describe('PublicRoute', () => {
  it('Should render transmitted component without auth', () => {
    const component = shallow(<PublicRoute authed={false} component={Dashboard} />);
    const routeProps = { location: 'string' };
    expect(component.find(Route).props().render(routeProps).type).toBe(Dashboard);
  });
  it('Should render without auth and without transmitted component', () => {
    const component = shallow(<PublicRoute authed={false} />);
    component.find(Route).props().render();
  });
  it('Should render Login without auth', () => {
    const component = shallow(<PublicRoute authed={true} />);
    const routeProps = { location: 'string' };
    expect(component.find(Route).props().render(routeProps).props.to.pathname).toBe(
      ROUTES.CALENDAR
    );
  });
});

describe('Router', () => {
  it('Should rendered with authed true', () => {
    const component = shallow(<MyRouter authed={true} />);
    expect(component).toMatchSnapshot();
  });

  it('Should rendered with authed false', () => {
    const component = shallow(<MyRouter authed={false} />);
    expect(component).toMatchSnapshot();
  });
});
