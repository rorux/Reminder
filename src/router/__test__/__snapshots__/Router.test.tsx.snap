// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router Should rendered with authed false 1`] = `
<Switch>
  <MainRoute
    authed={false}
    exact={true}
    path="/"
  />
  <PrivateRoute
    authed={false}
    exact={true}
    path="/calendar"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={false}
    exact={true}
    path="/dashboard"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={false}
    exact={true}
    path="/profile"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={false}
    exact={true}
    path="/create"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={false}
    exact={true}
    path="/edit/:id"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PublicRoute
    authed={false}
    exact={true}
    path="/login"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy />
    </Suspense>
  </PublicRoute>
  <PublicRoute
    authed={false}
    exact={true}
    path="/register"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy />
    </Suspense>
  </PublicRoute>
  <Route
    path="/not-found"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy />
    </Suspense>
  </Route>
  <Route>
    <Redirect
      to="/not-found"
    />
  </Route>
</Switch>
`;

exports[`Router Should rendered with authed true 1`] = `
<Switch>
  <MainRoute
    authed={true}
    exact={true}
    path="/"
  />
  <PrivateRoute
    authed={true}
    exact={true}
    path="/calendar"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={true}
    exact={true}
    path="/dashboard"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={true}
    exact={true}
    path="/profile"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={true}
    exact={true}
    path="/create"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PrivateRoute
    authed={true}
    exact={true}
    path="/edit/:id"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
      />
    </Suspense>
  </PrivateRoute>
  <PublicRoute
    authed={true}
    exact={true}
    path="/login"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy />
    </Suspense>
  </PublicRoute>
  <PublicRoute
    authed={true}
    exact={true}
    path="/register"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy />
    </Suspense>
  </PublicRoute>
  <Route
    path="/not-found"
  >
    <Suspense
      fallback={<ForwardRef(CircularProgress) />}
    >
      <lazy />
    </Suspense>
  </Route>
  <Route>
    <Redirect
      to="/not-found"
    />
  </Route>
</Switch>
`;
